pkg/controller/operators/olm/operator.go

```diff
@@ -1923,6 +1926,9 @@ func (a *Operator) getReplacementChain(in *v1alpha1.ClusterServiceVersion, csvsI
 		csvsInChain[current] = struct{}{}
 	}
 	for prev != nil && *prev != "" {
+		if _, ok := csvsInChain[*next]; ok {
```

Comment:
Should this be checking the previous csv in the chain?
```suggestion
		if _, ok := csvsInChain[*prev]; ok {
```

Reply:
Yes, good catch. Added a test case to cover this mistake and fixed it. Thanks!